### PR TITLE
Moved getlatestreleasetag function to meshkit

### LINKED ISSUE #131: [Enhancement] Avoid GitHub API Rate Limit
**Current Behavior**
Related to https://github.com/meshery/meshery/issues/4664, Meshery Adapters use the GH API to check for the latest version of a Meshery Adapter. Eventually, this will lead to an overage and GH API rate limit to be hit.

Example of Istio deployment operation failing due to overage:

```
./mesheryctl system logs meshery-istio
Starting Meshery logging...
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:39:01Z" level=info msg="Adaptor Listening at port: 10000" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:39:01Z" level=info msg="Could not get latest stable release" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:39:03Z" level=info msg="Creating instance" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:41:58Z" level=info msg="Creating instance" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:41:58Z" level=info msg="Listing Operations" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:41:58Z" level=info msg="Creating instance" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:42:35Z" level=info msg="Creating instance" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:42:38Z" level=info msg="Creating instance" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	time="2021-11-23T19:42:38Z" level=info msg="Applying operation istio" app=istio-adaptor
meshery-istio-54f78f78-gj4bp	|	panic: runtime error: index out of range [0] with length 0
meshery-istio-54f78f78-gj4bp	|
meshery-istio-54f78f78-gj4bp	|	goroutine 134 [running]:
meshery-istio-54f78f78-gj4bp	|	github.com/layer5io/meshery-istio/istio.(*Istio).ApplyOperation.func1(0xc00042c030, 0xc00081a0d0, 0x5, 0xc00081a0d5, 0x7, 0xc00081a0e0, 0x8, 0x0, 0x0, 0x0, ...)
meshery-istio-54f78f78-gj4bp	|		/build/istio/istio.go:51 +0x49a
meshery-istio-54f78f78-gj4bp	|	created by github.com/layer5io/meshery-istio/istio.(*Istio).ApplyOperation
meshery-istio-54f78f78-gj4bp	|		/build/istio/istio.go:50 +0x9ba
meshery-istio-54f78f78-gj4bp	|
```

**Desired Behavior**
Do not use api.github.com.

Use https://github.com/meshery/meshery/releases/latest

---
**Resources**

**Alternatives / Additional Context**

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #131 
This function can be used across meshery and all the adapters in order to remove  api.github.com calls
**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: utils/error.go [hunk 1]
```diff
@@ -67,3 +67,7 @@ func ErrReadingRemoteFile(err error) error {
 func ErrReadingLocalFile(err error) error {
 	return errors.New(ErrReadingLocalFileCode, errors.Alert, []string{"error reading local file"}, []string{err.Error()}, []string{"File doesnt exist in the location", "File name is incorrect"}, []string{"Make sure to input the right file name and location"})
 }
+
+func ErrGettingLatestReleaseTag(err error) error {
+	return errors.New(ErrGettingLatestReleaseTagCode, errors.Alert, []string{"Could not fetch latest stable release from github"}, []string{err.Error()}, []string{"Failed to make GET request to github", "Invalid response recieved on github.com/<org>/<repo>/releases/stable"}, []string{"Make sure Github is reachable", "Make sure a valid response is available on github.com/<org>/<repo>/releases/stable"})
```
Comment (leecalcote) [inline utils/error.go:72]: The use of "stable" here seems inaccurate. Will you have another think on this?

Comment (Revolyssup) [reply utils/error.go:72]: Yess, that was a mistake

**Review by piyushsingariya — APPROVED** [review]
LGTM!

File: utils/utils.go [hunk 2]
```diff
@@ -182,3 +185,34 @@ func ReadLocalFile(location string) (string, error) {
...
+		return "", ErrGettingLatestReleaseTag(err)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", ErrGettingLatestReleaseTag(err)
+	}
+	re := regexp.MustCompile("/releases/tag/(.*?)\"")
+	releases := re.FindAllString(string(body), -1)
+	latest := strings.ReplaceAll(releases[0], "/releases/tag/", "")
```
Comment (utkarsh-pro) [inline utils/utils.go:208]: Let's not assume that length > 0

Comment (piyushsingariya) [reply utils/utils.go:208]: Length will be always 1, but yeah better to add a if statement.

Comment (utkarsh-pro) [reply utils/utils.go:208]: @piyushsingariya what about the case when a repo has no releases?

Comment (leecalcote) [reply utils/utils.go:208]: Speaking of releases, the WASM filters need to be built and released - https://github.com/layer5io/wasm-filters/issues/38

Comment (leecalcote) [reply utils/utils.go:208]: This is a topic for today's Build and Release meeting.

**Review by utkarsh-pro — APPROVED** [review]

**Pull request merged**